components/Actuators/AutonomousControl/ballCollect.py

```diff
@@ -0,0 +1,92 @@
+from networktables import NetworkTables as networktable
+from components.SoftwareControl.speedSections import SpeedSections
+from components.Actuators.LowLevel.driveTrain import DriveTrain
+from components.Input.ballCounter import BallCounter
+from magicbot import StateMachine, state
+import logging as log
+
+class BallCollect(StateMachine):
+    compatString = ["teapot"]
+
+    speedSections: SpeedSections
+    driveTrain: DriveTrain
+    ballCounter: BallCounter
+
+    # In degrees
+    angleTolerance = .25
+    # In feet
+    distTolerance = 1
+    travelFirstCall = False
+
+    def startRunning(self):
+        self.start = True
+
+    def getBallDist(self):
+        # Placeholder, figure this out later.
+        return 5
+
+    @state(first=True)
+    def idling(self):
+        """
+        Starts the statemachine if requested.
+        Otherwise stays here.
+        """
+        self.next_state("idling")
+        if self.start:
+            self.start = False
+            self.forwardTurnSpeed = 0
+            self.next_state("align")

```

Comment:
Can startRunning() just do self.next_state("align") and no need to do a self.start = True? It would eliminate that variable completely right?